isb_web/sqlmodel_database.py

```diff
@@ -101,7 +107,12 @@ def read_things_summary(
 
 def get_thing_with_id(session: Session, identifier: str) -> Optional[Thing]:
     statement = select(Thing).filter(Thing.id == identifier)
-    return session.exec(statement).first()
+    result = session.exec(statement).first()
+    if result is None:
+        # Fall back to querying the Identifiers table
+        join_statement = select(Thing).join(ThingIdentifier).where(ThingIdentifier.guid == identifier)
+        result = session.exec(join_statement).first()
```

Comment:
These are the 4 lines or so that actually implement the feature.  The rest is all to support this 🥇